### PR TITLE
Update to Java 21 and Minecraft 1.21.2 - 1.21.4

### LINKED ISSUE #6: Crash on MC 1.12.2 - 1.12.4
Hi, do you plan to update this for 1.21.3/4? It doesn't prevent loading, but doing shift+ctrl+click to craft all crashes MC with the attached report
[crash-2024-12-13_00.37.40-client.txt](https://github.com/user-attachments/files/18119098/crash-2024-12-13_00.37.40-client.txt)

### PR DESCRIPTION
**Pull request by appelmoesgg**
Resolves #6 
It was bugged cause of some small fabric api changes.
This is fixed now

### CODE REVIEW

**Comment by 811Alex** [comment]
Thanks! Been kinda busy.
Have you tested this on any other MC versions?

**Comment by appelmoesgg** [comment]
This works on MC 1.21.2 and higher because they changed the API as you can see here:

[1.21](https://maven.fabricmc.net/docs/yarn-1.21+build.1/net/minecraft/screen/package-summary.html)
[1.21.2](https://maven.fabricmc.net/docs/yarn-1.21.2+build.1/net/minecraft/screen/package-summary.html)

In version 1.21.2, `AbstractCraftingScreenHandler` got added to the api, it took a part functionality of  `AbstractRecipeScreenHandler`, so I had to change some code to make it work.

For example: `AbstractRecipeScreenHandler.getCraftingResultSlotIndex()` got changed to `AbstractCraftingScreenHandler.getOutputSlot().getIndex()`

For versions under 1.21.2, the original code should still work...

File: build.gradle
Comment (appelmoesgg) [inline build.gradle:1]: Updated loom and java to newer versions

File: gradle.properties
Comment (appelmoesgg) [inline gradle.properties:1]: Updated dependencies for 1.21.4

File: gradle/wrapper/gradle-wrapper.properties
Comment (appelmoesgg) [inline gradle/wrapper/gradle-wrapper.properties:1]: Use newer gradle version (required for java 21)

File: src/main/java/eu/gflash/quickcraft/client/InventoryHelper.java
Comment (appelmoesgg) [inline src/main/java/eu/gflash/quickcraft/client/InventoryHelper.java:1]: [AbstractRecipeScreenHandler](https://maven.fabricmc.net/docs/yarn-1.21+build.1/net/minecraft/screen/AbstractRecipeScreenHandler.html) got changed to [AbstractCraftingScreenHandler](https://maven.fabricmc.net/docs/yarn-1.21.4+build.2/net/minecraft/screen/AbstractCraftingScreenHandler.html)

`AbstractRecipeScreenHandler.getCraftingResultSlotIndex ` got changed to `AbstractCraftingScreenHandler.getOutputSlot().getIndex()` (check javadocs to see why)

File: src/main/resources/quickcraft.mixins.json
Comment (appelmoesgg) [inline src/main/resources/quickcraft.mixins.json:1]: Java 21

File: src/main/resources/fabric.mod.json
Comment (appelmoesgg) [inline src/main/resources/fabric.mod.json:1]: Mention myself ofcourse :)

**Review by appelmoesgg — COMMENTED** [review]
Added some info to the file changes...

**Comment by 811Alex** [comment]
Alright, thanks for pointing out where they changed it.
I've already reviewed the rest and your commit messages were descriptive enough.

I appreciate the help!

**Pull request merged**